src/utils.tsx

```diff
@@ -48,8 +55,94 @@ export function getCookieValue(
         .map((c) => c.substring(name.length + 1))[0] ?? null
 }
 
-export function isLanguageCode(code: string): code is LanguageCode {
-  return Object.values(LanguageCode).some((x) => x === code)
+const PathInfo = t.type({ typename: t.string, currentPath: t.string })
+type PathInfo = t.TypeOf<typeof PathInfo>
+
+const ApiResult = t.type({
+  data: t.type({
+    uuid: t.intersection([
+      t.type({ __typename: t.string }),
+      t.partial({
+        alias: t.string,
+        pages: t.array(t.type({ alias: t.string })),
+      }),
+    ]),
+  }),
+})
+
+export async function getPathInfo(
+  lang: LanguageCode,
+  path: string
+): Promise<PathInfo | null> {
+  if (path.startsWith('/user/profile'))
+    return { typename: 'User', currentPath: path }
+
+  const cacheKey = `/${lang}${path}`
+  const cachedValue = await global.PATH_INFO_KV.get(cacheKey)
+
+  if (cachedValue !== null) {
+    try {
+      const result = PathInfo.decode(JSON.parse(cachedValue))
+
+      if (E.isRight(result)) return result.right
+    } catch (e) {
+      // ignore
+    }
+  }
+
+  const queryArg = /^\/\d+$/.test(path)
+    ? `id: ${path.slice(1)}`
+    : `alias: { instance: ${lang}, path: "${path}" }`
+  const query = `
+    {
+      uuid(${queryArg}) {
+        __typename
+        ... on Entity {
+          alias
+        }
+        ... on Course {
+          pages {
+            alias
+          }
+        }
+      }
+    }`
```

Comment:
Should use query + variables instead, see https://graphql.org/learn/queries/#variables.